apps/mobile/src/components/balance/balance.tsx

```diff
@@ -46,36 +43,42 @@ export function Balance({
   if (isLoading) {
     return <SkeletonLoader height={20} width={100} isLoading={true} />;
   }
-  const hasBalance = balance !== EmptyBalance && typeof balance !== 'string';
-  const isFiat = hasBalance && balance.symbol in currencyNameMap;
-  const balanceSymbol = hasBalance ? balance.symbol : undefined;
-  const maskedCurrencySymbol = !isFiat ? `*${balanceSymbol}` : undefined;
+  if (!balance) {
```

Comment:
if `balance === 0` won't this be truthy and show the "empty" balance? I think we do want to show `0` if it's actually 0, right?

Reply:
I'm open to improving this but I think it's OK? I've tested it out. 

`balance` is type `Money` so a `0` will come in as:
```
 balance {"amount": "0", "decimals": 8, "symbol": "BTC"}
 balance {"amount": "0", "decimals": 2, "symbol": "USD"}
```

And we will render `0` correctly

![Simulator Screenshot - iPhone 16 iOS 18 4 - 2025-05-16 at 05 23 31](https://github.com/user-attachments/assets/7263fadd-5620-4a7c-8d80-8b6dcb8b126c)

If we get an `error` the `balance.value` field of `fetchState` is `undefined` so it will be `(!balance)` and show the empty balance

![Simulator Screenshot - iPhone 16 iOS 18 4 - 2025-05-16 at 05 28 41](https://github.com/user-attachments/assets/5c9fff42-c34f-48e9-94ce-f33ee917d341)